kratos/python_scripts/multistage_orchestrators/multistage_orchestrator.py

```diff
@@ -0,0 +1,182 @@
+import sys
+import importlib
+
+import KratosMultiphysics
+from KratosMultiphysics.project import Project
+from KratosMultiphysics.model_parameters_factory import KratosModelParametersFactory
+
+class MultistageOrchestrator():
+    """Base class for multistage orchestrators.
+    
+    This class is intended to serve as base for al the Kratos multistage orchestrators.
+    The purpose of orchestrators is to define the workflow of the multistage simulations.
+    This includes deciding when and how to save and load (aka restart) the simulation.
+    Run() method must be implemented in derived classes (see SequentialMultistageOrchestrator).
+
+    Member variables:
+    __project -- Current multistage simulation project container
+    """
+
+    def __init__(self, project : Project) -> None:
+        """Constructs a multistage orchestrator instance"""
+
+        # Store pointer to current project
+        # Note that the project already contains the multistage simulation settings
+        self.__project = project
```

Comment:
```suggestion
        self.__project: Project = project
```

if you want to do it properly

Reply:
I wasn't aware of this. Thanks for the hint.